codersdk/externalauth.go

```diff
@@ -33,9 +34,12 @@ const (
 	EnhancedExternalAuthProviderAzureDevops EnhancedExternalAuthProvider = "azure-devops"
 	EnhancedExternalAuthProviderGitHub      EnhancedExternalAuthProvider = "github"
 	EnhancedExternalAuthProviderGitLab      EnhancedExternalAuthProvider = "gitlab"
-	EnhancedExternalAuthProviderBitBucket   EnhancedExternalAuthProvider = "bitbucket"
-	EnhancedExternalAuthProviderSlack       EnhancedExternalAuthProvider = "slack"
-	EnhancedExternalAuthProviderJFrog       EnhancedExternalAuthProvider = "jfrog"
+	// EnhancedExternalAuthProviderBitBucket is the Bitbucket Cloud provider.
+	// Not to be confused with the self-hosted 'EnhancedExternalAuthProviderBitBucketServer'
+	EnhancedExternalAuthProviderBitBucket       EnhancedExternalAuthProvider = "bitbucket"
```

Comment:
This will confuse people. Can we rename this to `EnhancedExternalAuthProviderBitBucketCloud` with `bitbucket-cloud` being the value, and in our `ParseEnvVars...` func just manually add a conversion for backwards compat?

Reply:
That works 👍. I was just worried about backwards compatibility.

It can't be done in ParseEnvVars though because the provider ID defaults to the provider type.